File: FSUIKit/Classes/FSTapScrollView.m
//
//  FSTapScrollView.m
//  ShareEconomy
//
//  Created by fudon on 16/6/8.
//  Copyright © 2016年 FudonFuchina. All rights reserved.
//

#import "FSTapScrollView.h"

@implementation FSTapScrollView

- (instancetype)initWithFrame:(CGRect)frame{
    self = [super initWithFrame:frame];
    if (self) {
        UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(tapActionInTapScrollView)];
        [self addGestureRecognizer:tap];
    }
    return self;
}

- (void)tapActionInTapScrollView{
    [self endEditing:YES];
}

/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/

@end

File: FSUIKit/Classes/Swift/FSTapScrollView.swift
// FSTapScrollViewS.swift
// Translated from FSTapScrollView.h/m

import UIKit

@objcMembers
public class FSTapScrollView: UIScrollView {
    
    public var click: ((FSTapScrollView) -> Void)?
    
    public override init(frame: CGRect) {
        super.init(frame: frame)
        setupTap()
    }
    
    required init?(coder: NSCoder) {
        super.init(coder: coder)
        setupTap()
    }
    
    private func setupTap() {
        let tap = UITapGestureRecognizer(target: self, action: #selector(tapActionInTapScrollView))
        addGestureRecognizer(tap)
    }
    
    @objc private func tapActionInTapScrollView() {
        click?(self)
    }
}
